### PR TITLE
v3.1.1 - Add the "limit" api parameter

### PR DESCRIPTION
**Pull request by thomktz**
## Methodology
Pseudo-code:
```
from flask_restx import Resource, reqparse
from . import api

# Parser for query parameters
parser = reqparse.RequestParser()
parser.add_argument('limit', type=int, required=False, default=50, help='Limit the number of lessons returned')

@api.route('/users/<int:id>/lessons')
class UserLessons(Resource):
    @api.expect(parser)
    def get(self, id):
        args = parser.parse_args()
        limit = args.get('limit')

        # use the limit in the function that gets the lessons...
        lessons = fetch_lessons_for_user(user_id=id, limit=limit)

        return lessons
```

### CODE REVIEW

**Comment by elenalmg** [comment]
- Add a parser limit in the route with default number of lessons/events returned = 50
image

- Modification of the GET lesson/event methods for a user in order to take into account the parameter limit. Example :
image

File: backend/app/operations/lesson_operations.py [hunk 1]
```diff
@@ -89,39 +89,50 @@ def get_lessons_by_class(class_: Class) -> list[Lesson]:
...
     """Get future lessons for a student through class groups."""
     # Retrieve all lessons for the student from their groups
     future_lessons = []
     date = datetime.now().date()
     for group in user.groups:
-        group_lessons = [lesson for lesson in group.lessons if lesson.date >= date]
+        all_lesson=group.lessons
+        group_lessons = [lesson for lesson in all_lesson if lesson.date >= date]
+
+    if limit is not None :
```
Comment (thomktz) [inline backend/app/operations/lesson_operations.py:115]: Don't think this should be here
`group_lessons` was defined in loop. Just use the limit in the final return statement

**Comment by elenalmg** [comment]
> Nice, few comments. Was it tested in the API? 

It has been tested in the API, everything works. 

> What should be the frontend syntax ?

The limit parameter, with a default value, is put directly in the request URL as you can see : 
image

**Comment by thomktz** [comment]
Added Changelog, and removed the limit on the calendar (you'd want to be able to see your calendar far into the future.)

**Pull request merged**